demo/src/execute.ts

```diff
@@ -24,6 +25,8 @@ export class AVRRunner {
   readonly portD: AVRIOPort;
   readonly usart: AVRUSART;
   readonly speed = 16e6; // 16 MHZ
+  readonly workUnitCycles = 500000;
+  readonly taskScheduler: MicroTaskScheduler;
```

Comment:
could be written as `readonly taskScheduler = new MicroTaskScheduler()`, so we don't have to repeat the type name in the constructor (but not a big deal to keep it as is, up to you)